### ISSUE TITLE
Electron 4.0.4 complains about NODE_MODULE_VERSION

### ISSUE THREAD

**Issue by kskalski** [grpc/grpc-node#762]
### Problem description
The grpc_node module seems to be compiled for NODE_MODULE_VERSION=64, but Electron 4 since 4.0.4 switched (https://github.com/electron/electron/pull/16687) to specifying NODE_MODULE_VERSION=69 and there is a mismatch at runtime trying to load the module.

### Reproduction steps
Run project with Electron 4.0.4 (or latest 4.0.7) and grpc npm dependency on 1.19.
The following error message is shown
![grpc_electron_node_module](https://user-images.githubusercontent.com/775012/53861258-3b4ef800-4016-11e9-9441-298b42472d4b.png)

### Environment
 - Windows 10, amd64
 - Electron 4.0.4
 - gRPC 1.19

### Additional context
Hopefully somewhere it's possible to specify the range of NODE_MODULE_VERSION accepted by the module... since it would be nice for the same module to work both with Electron 4.0.3 (which specifes NODE_MODULE_VERSION=64) and with newest Electron versions >=4.0.4

**Comment by nicolasnoble** [comment]
This is impossible with the current tools we have. Patch releases aren't supposed to break semantic versioning like this.

**Comment by kskalski** [comment]
True, I suppose the upgraded node module version was supposed to be there from the beginning and they just fixed it up in the patch release. Probably better to break compatibility with 4.0.0-4.0.3 and make it working with >=4.0.4 than what is there now.

**Comment by nicolasnoble** [comment]
This might be what we end up doing, yes. We'll discuss internally first.

**Comment by sla89** [comment]
Any update on this?

**Comment by nicolasnoble** [comment]
We've decided to wait for the Electron team to fix it on their end by publishing new packages.

Doing anything on our side makes little sense as we'd need to republish every single version of grpc that ever supported electron 4, whereas all they need to do is publish fixed packages.

**Comment by sla89** [comment]
FYI: https://github.com/electron/electron/releases/tag/v4.1.0

**Comment by kskalski** [comment]
So the next grpc release will have Electron packages for 4.1? I get this error:
```node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp WARN Tried to download(404): https://node-precompiled-binaries.grpc.io/grpc/v1.19.0/electron-v4.1-win32-ia32-unknown.tar.gz
node-pre-gyp WARN Pre-built binaries not found for grpc@1.19.0 and electron@4.1.0 (electron-v4.1 ABI, unknown) (falling back to source compile with node-gyp)```

**Comment by nicolasnoble** [comment]
Right.

In this case, we might even build 4.1 binaries for 1.19 right away. We'll update this later today.

**Comment by nicolasnoble** [comment]
We've been struggling with some compiler update which broke our build. We'll try posting 4.1 binaries when this is cleared up.

**Comment by HeHang0** [comment]
Hello， When will  4.1 binaries for 1.19  released?
Or 3.1 binaries for 1.19  released?，now I need 4.1 or 3.1. but I get 404 from https://node-precompiled-binaries.grpc.io/grpc/v1.19.0/electron-v3.1-darwin-x64-unknown.tar.gz

**Comment by CapOM** [comment]
I am facing the same issue, is there any solution ? I am trying to use electron 4.0.8 and grpc 1.19

**Comment by nicolasnoble** [comment]
The solution is to wait until we managed to get our compilation issues sorted.

**Comment by Slapbox** [comment]
@nicolasnoble is there any status update on this?

I'm not sure if it's related, but I believe it is; I'm suddenly seeing my Electron build size increase by 40% (a little over 30mb) all attributable to the `Build` folder inside of `node_modules/grpc.` It looks like in recent builds that folder didn't exist on my machine as best as I can tell.

**Comment by Djiit** [comment]
This is related; your machine builds its own binaries.
We just have to wait until they can solve their compilations issues.

**Comment by nicolasnoble** [comment]
We've finally sorted out the various issues plaguing our build, and we'll be releasing grpc 1.20 soon with Electron 3.1 and 4.1 prebuild binaries.

**Comment by Djiit** [comment]
Glad to here that. Thanks !

**Comment by nicolasnoble** [comment]
We just published grpc-1.20.0-pre1 with hopefully electron 3.1 and 4.1 binaries. We're no longer going to produce 4.0 binaries due to the ABI instability issue on this branch.

**Comment by kskalski** [comment]
thanks, works for me
